### PR TITLE
Implement suspendable method for the ISuspendable interface

### PR DESCRIPTION
**Pull request by matthias-wende-sociomantic**
Fixes https://github.com/sociomantic-tsunami/swarm/issues/391

### CODE REVIEW

File: src/swarm/neo/client/mixins/Controllers.d [hunk 1]
```diff
@@ -303,25 +311,23 @@ template Controllers ( )
...
-            throw new Exception("ISuspendable.suspended() is not currently supported "
-                ~ "by " ~ typeof(this).stringof);
+            this.control(
+                ( ControllerInterface controller )
+                {
+                    this._suspended = controller.suspended();
+                }
+            );
+
+            return this._suspended;
```
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/mixins/Controllers.d]: Does this need to be a class member? Could it not just be a local?

File: src/swarm/neo/client/mixins/Controllers.d [hunk 2]
```diff
@@ -303,25 +311,23 @@ template Controllers ( )
 
         /***********************************************************************
 
-            This method of `ISuspendable` cannot currently be implemented and
-            thus always throws. To implement this method correctly, this class
-            would need to somehow hook into the notifications of the request
-            being controlled, in order to receive the suspended/resumed
-            notifications to know when the request had actually changed state
-            (on the nodes' side).
+            TODO add description
```
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/mixins/Controllers.d]: What did you have in mind here? I'm not sure if we need to say anything beyond what Returns states now.

File: src/swarm/neo/client/mixins/Controllers.d [hunk 3]
```diff
@@ -281,6 +281,7 @@ template Controllers ( )
 
         ***********************************************************************/
 
+        deprecated("Only works with SuspendableRequestCore which is deprecated")
```
Comment (gavin-norman-sociomantic) [inline src/swarm/neo/client/mixins/Controllers.d:284]: Ah sorry, this will need an accompanying deprecation release note.

**Comment by matthias-wende-sociomantic** [comment]
Rebased and added release nodes.

**Review by gavin-norman-sociomantic — APPROVED** [review]

**Pull request merged**